packages/react/src/menu/popup/MenuPopup.tsx

```diff
@@ -107,7 +110,7 @@ export const MenuPopup = React.forwardRef(function MenuPopup(
       context={floatingContext}
       modal={false}
       disabled={!mounted}
-      returnFocus={finalFocus || returnFocus}
+      returnFocus={finalTriggerElement || returnFocus}
```

Comment:
This prevents `returnFocus` from ever being passed, no?

It seems like `returnFocus` needs to use the if branches above

Reply:
Included in this [commit](https://github.com/mui/base-ui/pull/2493/commits/2509ccf26579f268029d5538c20469560db085a5).

Reply:
Seems good now - also tested Menubar and the return focus issue this branch was addressing is still not present